diesel/src/query_source/aliasing/dsl_impls.rs

```diff
@@ -0,0 +1,269 @@
+use super::{Alias, AliasSource, FieldAliasMapper};
+
+use crate::associations::HasTable;
+use crate::dsl;
+#[cfg(feature = "postgres_backend")]
+use crate::expression::SelectableExpression;
+use crate::expression::{Expression, TypedExpressionType, ValidGrouping};
+use crate::expression_methods::EqAll;
+use crate::query_builder::{combination_clause, AsQuery, FromClause, Query, SelectStatement};
+use crate::query_dsl::methods::*;
+use crate::query_dsl::{CombineDsl, QueryDsl, RunQueryDsl};
+use crate::query_source::{QuerySource, Table};
+
+impl<S: AliasSource> QueryDsl for Alias<S> {}
+
+impl<S, Predicate> FilterDsl<Predicate> for Alias<S>
+where
+    Self: AsQuery,
+    <Self as AsQuery>::Query: FilterDsl<Predicate>,
+{
+    type Output = dsl::Filter<<Self as AsQuery>::Query, Predicate>;
+
+    fn filter(self, predicate: Predicate) -> Self::Output {
+        self.as_query().filter(predicate)
+    }
+}
+
+impl<S, Selection> SelectDsl<Selection> for Alias<S>
+where
+    Selection: Expression,
+    Self: AsQuery,
+    <Self as AsQuery>::Query: SelectDsl<Selection>,
+{
+    type Output = dsl::Select<<Self as AsQuery>::Query, Selection>;
+
+    fn select(self, selection: Selection) -> Self::Output {
+        self.as_query().select(selection)
+    }
+}
+
+impl<S, PK> FindDsl<PK> for Alias<S>
+where
+    S: AliasSource,
+    S::Table: Table + HasTable<Table = S::Table>,
+    <S::Table as Table>::PrimaryKey: FieldAliasMapper<S>,
+    <<S::Table as Table>::PrimaryKey as FieldAliasMapper<S>>::Out: EqAll<PK>,
+    Self: FilterDsl<
+        <<<S::Table as Table>::PrimaryKey as FieldAliasMapper<S>>::Out as EqAll<PK>>::Output,
+    >,
+{
+    type Output = dsl::Filter<
+        Self,
+        <<<S::Table as Table>::PrimaryKey as FieldAliasMapper<S>>::Out as EqAll<PK>>::Output,
```

Comment:
Looks like we should add a type alias to `diesel::dsl` for `<T as FieldAliasMapper<S>>::Out` and use that one here.

Reply:
So after having tried it in all places where `FieldAliasMapper` was used I feel like when used on generic types these typically make the link between the trait bounds `T: FieldAliasMapper<S>` and the other bounds or type constructions (`dsl::MappedFields<S, T>`) slightly less obvious hence harder to read, so I'd favor having the user use this alias with concrete types (so I've used it as signature of `Alias::fields`), but I haven't updated to using it for our internal trait bounds.

WDYT?

Reply:
Thats fine for me :+1: